packages/oidc-client/src/lib/authorize.request.ts

```diff
@@ -29,67 +27,141 @@ import type {
-         * updates, and want the request to be made similar to the iframe method below.
-         *
-         * This returns a Micro that resolves to the parsed response JSON.
-         */
-        return authorizeFetchµ(url).pipe(
-          Micro.flatMap(
-            (response): Micro.Micro<AuthorizeSuccessResponse, AuthorizeErrorResponse, never> => {
-              if ('code' in response) {
-                log.debug('Received code in response', response);
-                return Micro.succeed(response);
-              }
-              log.error('Error in authorize response', response);
-              // For redirection, we need to remove `pi.flow` from the options
-              const redirectOptions = options;
-              delete redirectOptions.responseMode;
-              return createAuthorizeErrorµ(response, wellknown, config, options);
-            },
-          ),
-        );
-      } else {
-        /**
-         * If the response mode is not pi.flow, then we are likely using a traditional
-         * redirect based server supporting iframes. An example would be PingAM.
-         *
-         * This returns a Micro that's either the success URL parameters or error URL
-         * parameters.
-         */
-        return authorizeIframeµ(url, config).pipe(
-          Micro.flatMap(
-            (response): Micro.Micro<AuthorizeSuccessResponse, AuthorizeErrorResponse, never> => {
-              if ('code' in response && 'state' in response) {
-                log.debug('Received authorization code', response);
-                return Micro.succeed(response as unknown as AuthorizeSuccessResponse);
-              }
-              log.error('Error in authorize response', response);
-              const errorResponse = response as unknown as AuthorizeErrorResponse;
-              return createAuthorizeErrorµ(errorResponse, wellknown, config, options);
-            },
-          ),
-        );
-      }
-    }),
+    Micro.flatMap(
+      ([url, options]): Micro.Micro<AuthorizationSuccess, AuthorizationError, never> => {
+        if (options.responseMode === 'pi.flow') {
+          /**
+           * If we support the pi.flow field, this means we are using a PingOne server.
+           * PingOne servers do not support redirection through iframes because they
+           * set iframe's to DENY.
+           *
+           * We do not use RTK Query for this because we don't want caching, or store
+           * updates, and want the request to be made similar to the iframe method below.
+           *
+           * This returns a Micro that resolves to the parsed response JSON.
+           */
+          return Micro.promise(() =>
+            store.dispatch(oidcApi.endpoints.authorizeFetch.initiate({ url })),
+          ).pipe(
+            Micro.flatMap(
+              (response): Micro.Micro<AuthorizationSuccess, AuthorizationError, never> => {
+                if ('error' in response) {
```

Comment:
This handles FetchBaseQueryError but not SerializedError. I have an example of handling both in the device client as recommended from the RTK docs.
https://github.com/ForgeRock/ping-javascript-sdk/blob/main/packages/device-client/src/lib/device.store.utils.ts

Reply:
Okay, I didn't do exactly what you suggested, but I added some handling for it.